src/utils/propertyCalculator.ts

```diff
@@ -177,6 +178,78 @@ export function useNewHealthFactorCalculator(
       ),
     [amount, tokenAddress, assetsData, collateral, increase]
   );
-
   return newHealthFactor;
 }
+
+export function useMaxChangeGivenHealthFactor(
+  amount: BigNumber | undefined,
+  tokenAddress: string | undefined,
+  mode: string,
```

Comment:
```suggestion
  mode: "repay" | "deposit" | "withdraw" | "borrow",
```
This will probably need you to do a bit of patching-up elsewhere if you do, but `string` is extremely vague and this only handles some string values.